[Community Event] Docstring Sprint

**Docstring** is important to understand what inputs a function/method expect and the output format it returns ! 

This issue is part of the **HACKTOBERFEST** event 🔥 . It is a call for contributions with the goal being to help `transformers` having the required and correct docstring, so users can  use it more smoothly.

Adding/Fixing a docstring is a simple (possibly first) contribution to Transformers and most importantly a very important contribution to the Transformers community ❤️  .

If you're interested in making a (maybe first!) contribution, please read through the **Guide to contributing** below. Before starting work on it, please reply in this thread which file you'd like to take :)

[An example of such PR](https://github.com/huggingface/transformers/pull/26636).

### Guide to contributing:
1. Ensure you've read our contributing [guidelines](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md) 📜 
2. Be sure you installed the development dependencies with `pip install -e ".[dev]"`, as described in the contributor guidelines above, to ensure that the code quality tools in `make fixup` can run.
3. Look at the file `utils/check_docstrings.py`:
    - find the line `OBJECTS_TO_IGNORE =` 
    - choose a name in the list `OBJECTS_TO_IGNORE ` (**make sure it is not taken yet by someone in the comments of this issue**)
    - **Claim the entry in this thread (confirm no one is working on it)** 🎯 
    - Let's select **one single** entry for a PR, or at most the entries from the same model architecture (its config/tokenizer/model/processor objects)
4. Remove the selected item (in step 3.) from `OBJECTS_TO_IGNORE`
    - commit the changes
5. run `python3 utils/check_docstrings.py --fix_and_overwrite`
    - You might see something like:
        - <img width="451" alt="Screenshot 2023-10-06 174057" src="https://github.com/huggingface/transformers/assets/2521628/750f241b-05e9-4970-8a20-f58a6d1192db">
    - commit the changes
6. fill information where `<fill_type>` or `<fill_docstring>` appear:
    - you can usually find the content to fill from other files (by searching the codebase)
    - compared to step 5.), the output now looks like:
        - <img width="305" alt="Screenshot 2023-10-06 182359" src="https://github.com/huggingface/transformers/assets/2521628/fa6c895e-ed56-468b-834c-490eefa0f954">

    - commit the changes
7. run `utils/check_docstrings.py`
    - make sure nothing change. Otherwise further work is required.
8. run `make fixup`
9. Open the PR 
    - with the title having format `[docstring] ... (entry name you work on) ...`
    - wait CI to be green
        - then tag me @ydshieh
    - otherwise, try to fix the failing tests if possible 🙏 . If necessary, ping me for helping on this.

**Looking forward to your contributions 🔥 ❤️ 🚀 !**

When I ran  `python3 -m pip install -e ".[dev-torch]" --user` command, I got below error:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
analytics-python 1.4.0 requires backoff==1.10.0, but you have backoff 1.11.1 which is incompatible.
fastai 2.7.12 requires torch<2.1,>=1.7, but you have torch 2.1.0 which is incompatible.
wandb 0.13.3 requires protobuf<4.0dev,>=3.12.0, but you have protobuf 4.24.4 which is incompatible.
```

After this, I ran `make fixup` and got below error:
```
No library .py files were modified
python utils/custom_init_isort.py
python utils/sort_auto_mappings.py
doc-builder style src/transformers docs/source --max_len 119 --path_to_docs docs/source
make: doc-builder: No such file or directory
make: *** [extra_style_checks] Error 1
```

Has second error caused because of first error?

> When I ran `python3 -m pip install -e ".[dev-torch]" --user` command, I got below error:
> 
> ```
> ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
> analytics-python 1.4.0 requires backoff==1.10.0, but you have backoff 1.11.1 which is incompatible.
> fastai 2.7.12 requires torch<2.1,>=1.7, but you have torch 2.1.0 which is incompatible.
> wandb 0.13.3 requires protobuf<4.0dev,>=3.12.0, but you have protobuf 4.24.4 which is incompatible.
> ```
> 
> After this, I ran `make fixup` and got below error:
> 
> ```
> No library .py files were modified
> python utils/custom_init_isort.py
> python utils/sort_auto_mappings.py
> doc-builder style src/transformers docs/source --max_len 119 --path_to_docs docs/source
> make: doc-builder: No such file or directory
> make: *** [extra_style_checks] Error 1
> ```
> 
> Is second error caused because of first error?
> 
> @ydshieh, @NielsRogge Could you please help me here. Sorry for asking many questions. I am new to transformers library and open source in general.

Maybe try with a new fresh python environment, preferably with miniconda and with a python version 3.8.x

--- a/src/transformers/models/canine/configuration_canine.py
+++ b/src/transformers/models/canine/configuration_canine.py
@@ -61,6 +61,9 @@ class CanineConfig(PretrainedConfig):
             The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
         layer_norm_eps (`float`, *optional*, defaults to 1e-12):
             The epsilon used by the layer normalization layers.
+        pad_token_id (`<fill_type>`, *optional*, defaults to 0): <fill_docstring>
+        bos_token_id (`<fill_type>`, *optional*, defaults to 57344): <fill_docstring>
+        eos_token_id (`<fill_type>`, *optional*, defaults to 57345): <fill_docstring>
         downsampling_rate (`int`, *optional*, defaults to 4):
             The rate at which to downsample the original character sequence length before applying the deep Transformer
             encoder.

--- a/src/transformers/models/canine/configuration_canine.py
+++ b/src/transformers/models/canine/configuration_canine.py
@@ -61,9 +61,12 @@ class CanineConfig(PretrainedConfig):
             The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
         layer_norm_eps (`float`, *optional*, defaults to 1e-12):
             The epsilon used by the layer normalization layers.
-        pad_token_id (`<fill_type>`, *optional*, defaults to 0): <fill_docstring>
-        bos_token_id (`<fill_type>`, *optional*, defaults to 57344): <fill_docstring>
-        eos_token_id (`<fill_type>`, *optional*, defaults to 57345): <fill_docstring>
+        pad_token_id (`int`, *optional*, defaults to 0):
+            Padding token id.
+        bos_token_id (`int`, *optional*, defaults to 57344):
+            Beginning of stream token id.
+        eos_token_id (`int`, *optional*, defaults to 57345):
+            End of stream token id.
         downsampling_rate (`int`, *optional*, defaults to 4):
             The rate at which to downsample the original character sequence length before applying the deep Transformer
             encoder.

--- a/utils/check_docstrings.py
+++ b/utils/check_docstrings.py
@@ -120,7 +120,6 @@ OBJECTS_TO_IGNORE = [
     "CamembertConfig",
     "CamembertModel",
     "CamembertTokenizerFast",
-    "CanineConfig",
     "CanineModel",
     "CanineTokenizer",
     "ChineseCLIPImageProcessor",

